onnxruntime/core/providers/cpu/tensor/concat.cc

```diff
@@ -35,66 +35,117 @@ Status ConcatBase::PrepareForCompute(OpKernelContext* ctx,
                                      const std::vector<const Tensor*>& input_tensors,
                                      Prepare& p) const {
   int input_count = static_cast<int>(input_tensors.size());
+
   // Must have atleast one input to concat
   ORT_RETURN_IF_NOT(input_count >= 1, "Must have 1 or more inputs");
 
-  const Tensor* tensor_pointer = input_tensors[0];
-  if (tensor_pointer == nullptr)
-    return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
+  std::vector<int64_t> reference_dims;
```

Comment:
side change: use the first non-empty tensor as the reference tensor to perform shape and rank validation on other input tensors. Previously, we just used the first one (which could be empty and hence we want to look for the first non-empty tensor) to keep as a reference